stake-pool/single-pool/src/processor.rs

```diff
@@ -0,0 +1,1079 @@
+            pool_mint_info.clone(),
+            pool_authority_info.clone(),
+            bump_seed,
+            token_amount,
+        )?;
+
+        // split stake into a blank stake account the user has created for this purpose
+        Self::stake_split(
+            vote_account_address,
+            pool_stake_info.clone(),
+            pool_authority_info.clone(),
+            bump_seed,
+            withdraw_stake,
+            user_stake_info.clone(),
+        )?;
+
+        // assign both authorities on the new stake account to the user
+        Self::stake_authorize(
+            vote_account_address,
+            user_stake_info.clone(),
+            pool_authority_info.clone(),
+            bump_seed,
+            user_stake_authority,
+            clock_info.clone(),
+            stake_program_info.clone(),
+        )?;
+
+        let post_pool_stake = get_active_stake_state(pool_stake_info, clock.epoch)?
+            .delegation
+            .stake;
+        msg!("Stake post split {}", post_pool_stake);
+
+        Ok(())
+    }
+
+    // this is an optional, but typical, part of initialization
+    // we separate it from the initialize instruction so that upstream cannot break us
+    #[inline(never)]
+    fn process_create_pool_token_metadata(
+        program_id: &Pubkey,
+        accounts: &[AccountInfo],
+        vote_account_address: &Pubkey,
+    ) -> ProgramResult {
+        let account_info_iter = &mut accounts.iter();
+        let pool_authority_info = next_account_info(account_info_iter)?;
+        let pool_mint_info = next_account_info(account_info_iter)?;
+        let payer_info = next_account_info(account_info_iter)?;
+        let metadata_info = next_account_info(account_info_iter)?;
+        let mpl_token_metadata_program_info = next_account_info(account_info_iter)?;
+        let system_program_info = next_account_info(account_info_iter)?;
+
+        let bump_seed = check_pool_authority_address(
+            program_id,
+            vote_account_address,
+            pool_authority_info.key,
+        )?;
+        check_pool_mint_address(program_id, vote_account_address, pool_mint_info.key)?;
+        check_system_program(system_program_info.key)?;
+        check_account_owner(payer_info, &system_program::id())?;
+        check_mpl_metadata_program(mpl_token_metadata_program_info.key)?;
+        check_mpl_metadata_account_address(metadata_info.key, pool_mint_info.key)?;
+
+        if !payer_info.is_signer {
+            msg!("Payer did not sign metadata creation");
+            return Err(SinglePoolError::SignatureMissing.into());
+        }
+
+        // checking the mint exists confirms pool is initialized
+        {
+            let pool_mint_data = pool_mint_info.try_borrow_data()?;
+            let _ = Mint::unpack_from_slice(&pool_mint_data)?;
+        }
+
+        let vote_address_str = vote_account_address.to_string();
+        let token_name = format!(
+            "SPL S-Val Token ({}...{})",
```

Comment:
nit: how about this exactly-32-characters name?
```suggestion
            "SPL Single StkPool ({}...{})",
```

Reply:
`"SPL Single Stake Pool {}..{}"`?

Reply:
and then maybe up here `"SPL Single Pool {}"` with as many bytes as we can fit